src/private/emscripten_api/emscripten_indoor_map_entity_information_api.js

```diff
@@ -63,77 +62,109 @@ function EmscriptenIndoorMapEntityInformationApi(emscriptenApiPointer, cwrap, ru
+            indoorMapEntityBufferSizesBuf.ptr);
+            
+        var indoorMapEntityBufferSize = _emscriptenMemory.consumeBufferToArray(indoorMapEntityBufferSizesBuf);
+        var indoorMapEntityStringIdSizes = _emscriptenMemory.consumeBufferToArray(indoorMapEntityIdsSizesBuf);
+        
+        var indoorMapEntityIdsTotalSize = indoorMapEntityBufferSize.shift();
+        var latLngsPerContourSize = indoorMapEntityBufferSize.shift();
+        var latLngsSize = indoorMapEntityBufferSize.shift();
+        
+        if (!success) {
+            return null;
+        }
+        
+        var indoorMapEntityStringIdsBuf = _emscriptenMemory.createInt8Buffer(indoorMapEntityIdsTotalSize);
+        
+        var indoorMapFloorIdBuf = _emscriptenMemory.createInt32Buffer(indoorMapEntityCount);
+        var positionLatLngBuf= _emscriptenMemory.createDoubleBuffer(indoorMapEntityCount * 2);
+
+        var contoursPerPolygonBuf = _emscriptenMemory.createInt32Buffer(indoorMapEntityCount);
+        var latLngsPerContourBuf = _emscriptenMemory.createInt32Buffer(latLngsPerContourSize);
+        var latLngsBuf = _emscriptenMemory.createDoubleBuffer(latLngsSize);
+        
+        success = _indoorEntityInformationApi_TryGetIndoorMapEntityModels(
+            _emscriptenApiPointer,
+            IndoorMapEntityInformationId,
+            indoorMapEntityCount,
+            indoorMapEntityStringIdsBuf.ptr,
+            indoorMapEntityIdsTotalSize,
+            indoorMapFloorIdBuf.ptr,
+            positionLatLngBuf.ptr,
+            contoursPerPolygonBuf.ptr,
+            latLngsPerContourBuf.ptr,
+            latLngsPerContourSize,
+            latLngsBuf.ptr,
+            latLngsSize
         );
-
-        var indoorMapEntityModelIds = _emscriptenMemory.consumeBufferToArray(indoorMapEntityModelIdsBuf);
-
+            
+        var indoorMapEntityStringIds = _emscriptenMemory.consumeUtf8BufferToString(indoorMapEntityStringIdsBuf);
+        var indoorMapEntityFloorIds = _emscriptenMemory.consumeBufferToArray(indoorMapFloorIdBuf);
+        var positionLatLng = _emscriptenMemory.consumeBufferToArray(positionLatLngBuf);
+        var contoursPerPolygon = _emscriptenMemory.consumeBufferToArray(contoursPerPolygonBuf);
+        var latLngsPerContour = _emscriptenMemory.consumeBufferToArray(latLngsPerContourBuf);
+        var latLngsDegrees = _emscriptenMemory.consumeBufferToArray(latLngsBuf);
+        
         if (!success)
         {
             return null;
         }
-
-        // Gets all indoorMapEntities from all Ids in indoorMapEntityModelIds
+        
         var indoorMapEntitiesList = [];
 
-        for (var entityIndex = 0; entityIndex < indoorMapEntityCount; ++entityIndex)
-        {
-            var indoorMapEntityStringIdSizeBuf = _emscriptenMemory.createInt32Buffer(1);
+        var idBufferHead = 0;
+        var latLngsPerContourHead = 0;
+        var latLngsDegreesHead = 0;
+        for (var i = 0; i < indoorMapEntityStringIdSizes.length; i++) {
```

Comment:
indoorMapEntityCount rather than indoorMapEntityStringIdSizes.length might be clearer (if I am understanding this correctly)